File: Python3/801 - 1000/999 Available Captures for Rook.py
# 36ms, 99.81%; 13.1MB, 100.00%
class Solution:
    def numRookCaptures(self, board: list) -> int:
        for i in range(8):
            if 'R' in board[i]:
                j = board[i].index('R')
                break
        res = 0
        for newi in range(i - 1, -1, -1):
            if board[newi][j] == '.': continue
            if board[newi][j] == 'p': res += 1
            break
        for newi in range(i + 1, 8):
            if board[newi][j] == '.': continue
            if board[newi][j] == 'p': res += 1
            break
        for newj in range(j - 1, -1, -1):
            if board[i][newj] == '.': continue
            if board[i][newj] == 'p': res += 1
            break
        for newj in range(j + 1, 8):
            if board[i][newj] == '.': continue
            if board[i][newj] == 'p': res += 1
            break
        return res
        

File: 801 - 1000/999 Available Captures for Rook.cc
// 4ms, 100.00%; 8.6MB, 100.00%
#include <bits/stdc++.h>
using namespace std;

static int x = []() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);
    return 0;
}();

class Solution {
public:
    int numRookCaptures(const vector<vector<char>> &board) {
        int i = 0, j = 0;
        while (true) {
            if (board[i][j] == 'R') break;
            (j == 7) ? ++i, j = 0 : ++j;
        }
        int res = 0;
        for (int newi = i - 1; newi >= 0; --newi) {
            if (board[newi][j] == '.') continue;
            if (board[newi][j] == 'p') ++res;
            break;
        }
        for (int newi = i + 1; newi < 8; ++newi) {
            if (board[newi][j] == '.') continue;
            if (board[newi][j] == 'p') ++res;
            break;
        }
        for (int newj = j - 1; newj >= 0; --newj) {
            if (board[i][newj] == '.') continue;
            if (board[i][newj] == 'p') ++res;
            break;
        }
        for (int newj = j + 1; newj < 8; ++newj) {
            if (board[i][newj] == '.') continue;
            if (board[i][newj] == 'p') ++res;
            break;
        }
        return res;
    }
};
